### DISCUSSION TITLE
Sharing global data through app

### DISCUSSION THREAD

**Discussion by vadym-oliinyk** [withastro/roadmap#345] [Proposal]
Hi, what is the best way to share global **constant** data through an app?

For example, I have a configuration object received from API. Both server (Astro) and clients (Preact) components need this data for read-only purposes. 
Should I use [state sharing ](https://docs.astro.build/en/core-concepts/sharing-state/)? (because, in fact its immutable data)
Or maybe create a global constant (`global.config`) and share it with the client through the inline script (to prevent callback hell between components)?
Or there are better ways to achieve these?

P.S. The same question relates to sharing translation messages through an app.
P.S.S. Thanks for your help. I`m new to Astro, so I may not know about the obvious solutions.

**Comment by jasikpark** [comment]
Maybe use something like https://github.com/natemoo-re/ultrafetch or https://tanstack.com/query/v4/docs/overview? So you can have caching on build? Do you need this to be updated at build time or also at page-load time?

**Reply by vadym-oliinyk** [reply]
Thanks for your answer. I got SSR, so need this data on runtime

**Comment by florian-lefebvre** [comment]
Closing because this is outdated
